pkg/osbuild/bootupd_stage.go

```diff
@@ -130,6 +143,26 @@ func GenBootupdDevicesMounts(filename string, pt *disk.PartitionTable, pltf plat
 			},
 		},
 	}
+	for partNum0, part := range pt.Partitions {
```

Comment:
I don't understand the `0` suffix... Is that to make it obvious that the index starts at 0? I suggest putting a little comment here (or before `partNum0 + 1`) to make it obvious.

Reply:
It was meant as a little reminder that zero-indexed but we need a one-indexed result but if it's more confusing than helping I remove it pagain and just call it `idx` :)

Reply:
Then my understanding was correct! :) I suggest adding an explicit reminder via a comment then. :)